### DISCUSSION TITLE
S57 (ENC) support?

### DISCUSSION THREAD

**Discussion by yunusyurtturk** [pelicanmapping/osgearth#2636] [Support Q&A] [answered]
Hello;
Does osgEarth support the S57 file format (S57 - Electronic Navigation Charts - ENC)? I couldn't find any 's57' or 'ENC' driver for osgEarth, or any examples. Are there any available?

I noticed that GDAL/OGR can read the S57 format (if S57 support is enabled, which is supposed to be the default). So, I started looking into the Boston city buildings example, which uses OGRFeatureSource. If there aren't any examples for the S57 format, could I implement some basic S57 support using OGRFeatureSource and related classes? Or if there are better ways or examples to follow, please guide me to the right direction.

**Comment by yunusyurtturk** [comment]
This is the gdal doc about S57: https://gdal.org/en/latest/drivers/vector/s57.html

I am unable to read S57 features with this method btw:

```
osg::ref_ptr<FeatureCursor> cursor = ogrFeatureSource->createFeatureCursor();
while(cursor.valid() && cursor->hasMore()){
    
}
```

**Comment by gwaldron** [comment, marked as answer]
S57 is comprised of many layers. The `OGRFeatureSource` can only open one layer at a time, so have to specify the layer. Make sure you set it before opening the layer:
```c++
ogrFeatureSource->setLayer("SEAARE");
```
And in an earth file
```xml
<OGRFeatures>
    <url>US2EC03M.000</url>
    <layer>SEAARE</layer>
</OGRFeatures>
```

**Comment by yunusyurtturk** [comment]
Great, now I'm able to get the features & understand the process. Thank you!
